### ISSUE TITLE
Backward propagation safety, Loid initialization

### ISSUE THREAD

**Issue by smirarab** [puoya/htree#12]
This line may not be backward propagation safe:

https://github.com/puoya/htree/blob/110f841b610fbe54ea9ec36c7360331f1aee3d32/htree/embedding.py#L663

I think it can cause a `one of the variables needed for gradient computation has been modified by an inplace operation` error. I am not sure. 

Perhaps something along the lines of this would work?

~~~bash
space_parts = points[1:, :] 

time_parts = torch.sqrt(1 + torch.sum(space_parts**2, dim=0, keepdim=True) )

points = torch.cat([time_parts, space_parts], dim=0)
~~~
